src/OmniSharp.Roslyn.CSharp/Services/Navigation/GotoDefinitionService.cs

```diff
@@ -31,8 +31,12 @@ public GotoDefinitionService(OmniSharpWorkspace workspace, MetadataHelper metada
         {
             var quickFixes = new List<QuickFix>();
 
-            var document = _workspace.GetDocument(request.FileName);
+            var document = request.IsMetadataFile() ? 
+                _metadataHelper.FindDocumentInMetadataCache(request.FileName) : 
+                _workspace.GetDocument(request.FileName);
```

Comment:
This could technically work for other endpoints as well right?  Thinking of `/findusages`, `/typelookup` and maybe even `/highlight`.

Would be great if we could simplify this.